tools/cli/commands/connect.py

```diff
@@ -189,13 +190,12 @@ def ensure_sshable():
             '--ssh-flag=LogLevel=' + args.ssh_log_level])
         cmd.append('datalab@{0}'.format(instance))
         cmd.extend(['--', 'true'])
-        with open(os.devnull, 'w') as dn:
-            while True:
-                try:
-                    gcloud_compute(args, cmd, stderr=dn)
-                    return
-                except subprocess.CalledProcessError:
-                    pass
+        for attempt in range(max_ssh_attempts):
+            try:
+                gcloud_compute(args, cmd)
+                return
+            except subprocess.CalledProcessError:
+                pass
```

Comment:
This isn't quite right.

There's nothing that signals an error if the attempt limit is hit.